### PR TITLE
Fixes Coverity NullReferenceException Issues

### PR DESCRIPTION
**Pull request by whoisj**
The new-again Coverity scanning has started to bear fruit. Here are some of the NRE issues identified by the scan. Minor fixes, low risk. **Not sure if I'm throwing the correct exception types** here (/CC @nulltoken).

1. Do not assume disk reads always succeed, check results before processing.
2. Verify parameters being using.
3. There's more than one kind of `null` when dealing with p/invoke and handles.
4. Better to give specific exceptions than general exceptions when reading in bad data.

### CODE REVIEW

**Comment by nulltoken** [comment]
@whoisj I think `Version.cs` is already targeted by #1080

File: LibGit2Sharp/ObjectDatabase.cs [hunk 1]
```diff
@@ -251,7 +251,7 @@ public virtual Blob CreateBlob(Stream stream, long numberOfBytesToConsume)
 
             using (var odbStream = Proxy.git_odb_open_wstream(handle, numberOfBytesToConsume, GitObjectType.Blob))
             {
-                var buffer = new byte[4*1024];
+                var buffer = new byte[4 * 1024];
```
Comment (nulltoken) [inline LibGit2Sharp/ObjectDatabase.cs]: Oh.... Look Ma... Spaces! :trollface:

Comment (whoisj) [inline LibGit2Sharp/ObjectDatabase.cs]: Spaces are bad? Not sure what you mean here. Is it just that the Roslyn autoformatter struck again?

File: LibGit2Sharp/ObjectDatabase.cs [hunk 2]
```diff
@@ -365,7 +365,9 @@ public virtual Commit CreateCommit(Signature author, Signature committer, string
 
             ObjectId commitId = Proxy.git_commit_create(repo.Handle, null, author, committer, message, tree, parentIds);
 
-            return repo.Lookup<Commit>(commitId);
+            Commit commit = repo.Lookup<Commit>(commitId);
+            Ensure.GitObjectIsNotNull(commit, "commit");
```
Comment (nulltoken) [inline LibGit2Sharp/ObjectDatabase.cs]: I think something like `Ensure.GitObjectIsNotNull(commit, commitId.Sha);` may be more helpful when troubleshooting.

File: LibGit2Sharp/ObjectDatabase.cs [hunk 3]
```diff
@@ -454,6 +456,8 @@ public virtual string ShortenObjectId(GitObject gitObject)
         /// <returns>A short string representation of the <see cref="ObjectId"/>.</returns>
         public virtual string ShortenObjectId(GitObject gitObject, int minLength)
         {
+            Ensure.GitObjectIsNotNull(gitObject, "gitObject");
```
Comment (nulltoken) [inline LibGit2Sharp/ObjectDatabase.cs]: `Ensure.ArgumentNotNull(gitObject, "gitObject");` should be enough here.

File: LibGit2Sharp/ObjectDatabase.cs [hunk 4]
```diff
@@ -462,6 +466,11 @@ public virtual string ShortenObjectId(GitObject gitObject, int minLength)
 
             string shortSha = Proxy.git_object_short_id(repo.Handle, gitObject.Id);
 
+            if (shortSha == null)
+            {
+                throw new LibGit2SharpException("Unable to obtain SHA-1 value for GitObject " + gitObject.Id);
```
Comment (nulltoken) [inline LibGit2Sharp/ObjectDatabase.cs]: s/obtain/abbreviate/ ?

File: LibGit2Sharp/ObjectDatabase.cs (hunk 1 above)
Comment (nulltoken) [inline LibGit2Sharp/ObjectDatabase.cs]: >Is it just that the Roslyn autoformatter struck again?

@whoisj I'm not against spaces :wink: However, I'd rather keep (unrelated) beautification in separate commits.

Comment (whoisj) [inline LibGit2Sharp/ObjectDatabase.cs]: Ah gotcha - @#$%! Roslyn and her free beautification on every `}` and `;` :grin:

**Comment by whoisj** [comment]
@nulltoken fixed up. Removed my fixes for Version.cs

**Comment by nulltoken** [comment]
:8ball:

**Pull request merged**